src/libraries/Common/src/System/Security/Cryptography/HashOneShotHelpers.cs

```diff
@@ -28,6 +28,18 @@ internal static byte[] HashData(HashAlgorithmName hashAlgorithm, ReadOnlySpan<by
             {
                 return SHA384.HashData(source);
             }
+            else if (hashAlgorithm == HashAlgorithmName.SHA3_256)
+            {
+                return SHA3_256.IsSupported ? SHA3_256.HashData(source) : throw new PlatformNotSupportedException();
+            }
+            else if (hashAlgorithm == HashAlgorithmName.SHA3_384)
+            {
+                return SHA3_384.IsSupported ? SHA3_384.HashData(source) : throw new PlatformNotSupportedException();
+            }
+            else if (hashAlgorithm == HashAlgorithmName.SHA3_512)
+            {
+                return SHA3_512.IsSupported ? SHA3_512.HashData(source) : throw new PlatformNotSupportedException();
+            }
             else if (Helpers.HasMD5 && hashAlgorithm == HashAlgorithmName.MD5)
```

Comment:
Should we switch to using a switch on the name now that the list has grown considerably?

Reply:
> Should we switch to using a switch on the name now that the list has grown considerably?

We can. Do you think that will be a meaningful performance improvement, or are you suggesting it as a stylistic change?

Reply:
Mainly style / consistency.  This is a much more verbose way of expressing the options than is used elsewhere.

Reply:
Ah, but now I see why they aren't a switch - these aren't strings. These are `HashAlgorithmName`.

Reply:
Right, but can't you just switch on .Name?

Reply:
Yeah, I was mostly handwringing about where to get the case values from, but we have costs declared elsewhere that we can use.

We can also eliminate the `IsSupported` check because the one-shot is going to do that check anyway and throw the PNSE.